desktop_version/src/Map.cpp

```diff
@@ -396,6 +398,59 @@ void mapclass::transformname(int t)
     }
 }
 
+void mapclass::updateroomnames(void)
+{
+    if (roomnameset)
+    {
+        return;
+    }
+
+    const int rx = game.roomx - 100;
+    const int ry = game.roomy - 100;
+
+    for (int i = specialroomnames.size() - 1; i >= 0; i--)
+    {
+        Roomname &roomname = specialroomnames[i];
```

Comment:
Use a regular pointer instead of a C++ reference. Trying to limit the amount of C++isms used in the codebase...